projects/plugins/jetpack/modules/widgets/migrate-to-core/gallery-widget.php

```diff
@@ -92,81 +107,93 @@ function jetpack_migrate_gallery_widget_is_importable( $widget ) {
 		return null;
 	}
 	// Ensure widget has no keys other than those expected.
 	// Not all widgets have conditions, so lets add it in.
-	$widget_copy = array_merge( array( 'conditions' => null ), $widget );
+	$widget_copy      = array_merge( array( 'conditions' => null ), $widget );
 	$non_allowed_keys = array_diff_key( $widget_copy, $allowed_keys );
 	if ( count( $non_allowed_keys ) > 0 ) {
 		jetpack_migrate_gallery_widget_bump_stats( 'extra-key' );
 
 		// Log the names of the keys not in our allowed list.
 		foreach ( $non_allowed_keys as $key => $value ) {
-			jetpack_migrate_gallery_widget_bump_stats( "extra-key-$key", "migration-extra-key" );
+			jetpack_migrate_gallery_widget_bump_stats( "extra-key-$key", 'migration-extra-key' );
 		}
 	}
 
-	$widget_copy = array_merge( $default_data, $widget, array(
-		// ids in Jetpack's Gallery are a string of comma-separated values.
-		// Core's Media Gallery Widget stores ids in an array
-		'ids'            => explode( ',', $widget['ids'] ),
-		'link_type'      => $widget['link'],
-		'orderby_random' => isset( $widget['random'] ) && $widget['random'] === 'on',
-	) );
+	$widget_copy = array_merge(
+		$default_data,
+		$widget,
+		array(
+			// ids in Jetpack's Gallery are a string of comma-separated values.
+			// Core's Media Gallery Widget stores ids in an array.
+			'ids'            => explode( ',', $widget['ids'] ),
+			'link_type'      => $widget['link'],
+			'orderby_random' => isset( $widget['random'] ) && 'on' === $widget['random'],
+		)
+	);
 
-	// Unsetting old widget fields
-	$widget_copy = array_diff_key( $widget_copy, array(
-		'link' => false,
-		'random' => false,
-	) );
+	// Unsetting old widget fields.
+	$widget_copy = array_diff_key(
+		$widget_copy,
+		array(
+			'link'   => false,
+			'random' => false,
+		)
+	);
 
 	return $widget_copy;
 }
 
 /**
  * Replaces the references to Jetpack Gallery Widget in the sidebars for references to the new version of the widget
  *
- * @param $sidebars_widgets   The sidebar widgets array to update
- * @param $id                 Old id of the widget (basically its index in the array )
- * @param $new_id             New id that will be using on the sidebar as a new widget
+ * @param array  $sidebars_widgets The sidebar widgets array to update.
+ * @param string $id               Old id of the widget (basically its index in the array ).
+ * @param string $new_id           New id that will be using on the sidebar as a new widget.
  *
- * @return mixed              Updated sidebar widgets array
+ * @return mixed                   Updated sidebar widgets array
  */
 function jetpack_migrate_gallery_widget_update_sidebars( $sidebars_widgets, $id, $new_id ) {
 	foreach ( $sidebars_widgets as $sidebar => $widgets ) {
+		$key = is_array( $widgets ) ? array_search( "gallery-{$id}", $widgets, true ) : false;
+
 		if (
 			is_array( $widgets )
-			&& false !== ( $key = array_search( "gallery-{$id}", $widgets, true ) )
+			&& false !== $key
 		) {
```

Comment:
No need to check `is_array()` again now.
```suggestion
		if ( false !== $key ) {
```

Reply:
👍 b7c781e90394caa4c851f6a770e99d7648721e90